### PR TITLE
KPIValue integration tests

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard1**
## Summary

Add integration tests for all KPIValue instances to check if they're displaying the correct data key.

## Motivation

Testing is good

## Detailed design

Not much to explain really. I've re-used the data-cy attribute to mark the kpi values, these were already used by the Cypress tests, so in order not to riddle the code with too many testing related attributes I chose to use the same one.

## Drawbacks

n/a

## Alternatives

n/a

## Unresolved questions

n/a

### Governance

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

**Review by VWSCoronaDashboard8 — APPROVED** [review]

File: src/pages-tests/gemeente/[code]/rioolwater.test.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    );
+  });
+
+  it('should use total_installation_count for total installation count', () => {
+    testKpiValue(
+      container,
+      'total_installation_count',
+      formatNumber(data.sewer?.last_value.total_installation_count)
+    );
+  });
```
Comment (VWSCoronaDashboard10) [inline src/pages-tests/gemeente/[code]/rioolwater.test.tsx:95]: Can't we test all kpi values in a single test?
eg:
```tsx
it('should render correct data', () => {
  testKpiValues(
    container,
    {
      total_installation_count: formatNumber(data.sewer?.last_value.total_installation_count),
      barscale_value: formatNumber(dataCopy.sewer_per_installation.values[0].last_value.rna_normalized),
      et: formatNumber(cet),
      e: formatNumber(ra),
      // ...etc
    }
  );
});

File: src/pages-tests/landelijk/positief-geteste.mensen.test.tsx [hunk 2]
```diff
@@ -1,49 +1,44 @@
...
+  ) => HTMLElement[];
+
+  beforeEach(() => {
+    const renderResult = render(
+      <PositiefGetesteMensen data={data} lastGenerated="test" />
+    );
+    container = renderResult.container;
+    getByText = renderResult.getByText;
+    getAllByRole = renderResult.getAllByRole;
+  });
```
Comment (VWSCoronaDashboard10) [inline src/pages-tests/landelijk/positief-geteste.mensen.test.tsx:27]: What's the reason of all the separate assignments? How about only assigning the result of `render` to a shared variable?
```tsx
let result: RenderResult;

beforeEach(() => {
  result = render(
    <PositiefGetesteMensen data={data} lastGenerated="test" />
  )
})

// ...

result.getByText('...')
```

Comment (VWSCoronaDashboard1) [reply src/pages-tests/landelijk/positief-geteste.mensen.test.tsx:27]: fair enough ;)

File: src/pages-tests/gemeente/[code]/rioolwater.test.tsx (hunk 1 above)
Comment (VWSCoronaDashboard1) [reply src/pages-tests/gemeente/[code]/rioolwater.test.tsx:95]: True, I prefer I bit of verbosity when it comes to tests though. Plus the 'one expectation per test', makes things immediately clear when a test fails. When you're building an elaborate test that loops through an entire manually constructed dataset it doesn't necessarily make the results instantly readable. IMHO of course...

File: src/pages-tests/gemeente/[code]/rioolwater.test.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            },
+          },
+        ],
+      };
+    } else {
+      const firstInstallation = dataCopy.sewer_per_installation?.values[0];
+      dataCopy.sewer_per_installation.values = [firstInstallation];
+    }
+
+    const { container } = render(
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard1) [reply src/pages-tests/gemeente/[code]/rioolwater.test.tsx:72]: fixed

File: src/test-utils/test-kip-value.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { getTextByDataCy } from './get-text-by-data-cy';
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard1) [reply src/test-utils/test-kip-value.ts:1]: fixed

File: src/pages-tests/gemeente/[code]/rioolwater.test.tsx (hunk 1 above)
Comment (VWSCoronaDashboard10) [reply src/pages-tests/gemeente/[code]/rioolwater.test.tsx:95]: Ok feel free to implement what feels right, I might not have enough experience with tests and its results to have a reasoned opinion about this :-)

Comment (VWSCoronaDashboard1) [reply src/pages-tests/gemeente/[code]/rioolwater.test.tsx:95]: Opinions vary, I never mind the discussion though, it's interesting to hear other viewpoints

**Pull request merged**